File: dog.js
var Dog = function(args){
	this.name = args.name || "Sunshine";
	this.age = args.age;
	this.breed = args.breed;
	// this.bark = function(){
	// 	return "bark";
	// }
}

Dog.prototype.bark = function(){
	return "bark";
}

var dog_vars = { age: "10", breed: "Pugle"}
var dog = new Dog(dog_vars)
console.log(dog.bark())

File: dog.rb
class Dog

  def initialize(args={})
    @name = args.fetch("name", "Sally")
    @age = args.fetch("age", "0")
    @breed = args.fetch("breed", "Dalmation")
  end

  def bark
    puts "bark"
  end
end

puts dog = Dog.new(name: "hello")

dog.bark